### PR TITLE
Feature/tls runtime set

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

Slow and fragile TLS setup.
[DPE-1447](https://warthogs.atlassian.net/browse/DPE-1447)

## Solution

Port TLS hot reload with lib from merged [PR#142](https://github.com/canonical/mysql-operator/pull/142)

[DPE-1447]: https://warthogs.atlassian.net/browse/DPE-1447?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shayancanonical — COMMENTED** [review]
Should we delete the rolling ops lib since we are no longer using it?

**Comment by paulomach** [comment]
> Should we delete the rolling ops lib since we are no longer using it?

what a miss, thanks!
Done db310a1

**Review by taurus-forever — APPROVED** [review]
Tnx!

File: src/charm.py [hunk 1]
```diff
@@ -192,7 +187,9 @@ def get_unit_hostname(self, unit_name: str) -> str:
         Returns:
             A string representing the hostname.localdomain of the unit.
         """
-        return f"{unit_name.replace('/', '-')}.{self.app.name}-endpoints"
+        if unit_name:
+            return f"{unit_name.replace('/', '-')}.{self.app.name}-endpoints"
+        return f"{self.unit.name.replace('/', '-')}.{self.app.name}-endpoints"
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/charm.py]: 0fbf8d0

**Pull request merged**